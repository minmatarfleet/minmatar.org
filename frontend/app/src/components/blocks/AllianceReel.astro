---
import { i18n } from '@helpers/i18n'
const { t, translatePath } = i18n(Astro.url)

interface Props {
    can_view_applications?: boolean;
    user_is_officer?:       boolean;
    user_is_director?:      boolean;
    can_view_corp_members?: boolean;
    is_superuser?:          boolean;
}

const {
    can_view_applications = false,
    user_is_officer = false,
    user_is_director = false,
    can_view_corp_members = false,
    is_superuser = false,
} = Astro.props

import FlexInline from '@components/compositions/FlexInline.astro';
import Wrapper from '@components/compositions/Wrapper.astro';

import LandingSwiper from '@components/blocks/LandingSwiper.astro';
import Button from '@components/blocks/Button.astro';
import SwiperCard from '@components/blocks/SwiperCard.astro';
import Tag from '@components/blocks/Tag.astro';
import Markdown from '@components/blocks/Markdown.astro';

import AugmentationEvEIcon from '@components/icons/AugmentationEvEIcon.astro';
import GroupsEvEIcon from '@components/icons/GroupsEvEIcon.astro';
import CorporationEvEIcon from '@components/icons/CorporationEvEIcon.astro';
---

<Wrapper
    class="[ alliance-reel ]"
    padding_block='0'
    padding_inline='0'
    max_width='100%'
>
    <div
        class="[ swiper ][ w-full pb-10 ]"
        id="corporations-list"
    >
        <LandingSwiper>
            <div class="[ swiper-slide ]" style="margin-right: 40px">
                <SwiperCard class="[ animate-delay-0 ]" cover='/images/corporations-tile-background.jpg'>                    
                    <FlexInline slot="header" gap='var(--space-2xs)'>
                        <CorporationEvEIcon />
                        <h2>{t('corporations')}</h2>
                    </FlexInline>
                        
                    <p>{t('corporations.leading_text')}</p>
                    <FlexInline>
                        <Button size='sm' href={translatePath('/alliance/corporations/list/')}>
                            {t('view')}
                        </Button>
                        {can_view_applications &&
                            <Button size='sm' href={translatePath('/alliance/corporations/applications/')}>
                                {t('view_applications')}
                            </Button>
                        }
                        {can_view_corp_members &&
                            <Button size='sm' href={translatePath('/alliance/corporations/members/')}>
                                {t('view_members')}
                            </Button>
                        }
                    </FlexInline>
                </SwiperCard>
            </div>

            <div class="[ swiper-slide ]" style="margin-right: 40px">
                <SwiperCard class="[ animate-delay-200 ]" cover='/images/sigs-tile-background.jpg'>                    
                    <FlexInline slot='header' gap='var(--space-2xs)'>
                        <GroupsEvEIcon />
                        <h2>{t('groups.list.page_title')}</h2>
                    </FlexInline>
                        
                    <Markdown markdown={t('groups.leading_text_p2')} inline={true} />
                    <FlexInline>
                        <Button size='sm' href={translatePath('/alliance/sigs/list/')}>
                            {t('view_groups')}
                        </Button>
                        
                        {user_is_officer === null &&
                            <Tag text={t('page_reload_error')} />
                        }
                        {(user_is_officer || is_superuser) &&
                            <Button size='sm' href={translatePath('/alliance/sigs/requests/')}>
                                {t('group_requests')}
                            </Button>
                            
                            <Button size='sm' href={translatePath('/alliance/sigs/members/')}>
                                {t('view_members')}
                            </Button>
                        }
                    </FlexInline>
                </SwiperCard>
            </div>
            
            <div class="[ swiper-slide ]">
                <SwiperCard class="[ animate-delay-400 ]" cover='/images/teams-tile-background.jpg'>
                    <FlexInline slot='header' gap='var(--space-2xs)'>
                        <AugmentationEvEIcon />
                        <h2>{t('teams.list.page_title')}</h2>
                    </FlexInline>
                        
                    <p>{t('teams.leading_text')}</p>
                    <FlexInline>
                        <Button size='sm' href={translatePath('/alliance/teams/list/')}>
                            {t('view_teams')}
                        </Button>
                        
                        {user_is_director === null &&
                            <Tag text={t('page_reload_error')} />
                        }
                        {(user_is_director || is_superuser) &&
                            <Button size='sm' href={translatePath('/alliance/teams/requests/')}>
                                {t('team_requests')}
                            </Button>
                            
                            <Button size='sm' href={translatePath('/alliance/teams/members/')}>
                                {t('view_members')}
                            </Button>
                        }
                    </FlexInline>
                </SwiperCard>
            </div>
        </LandingSwiper>
    </div>

    <div class="[ swiper-button-prev hidden lg:block ]"></div>
    <div class="[ swiper-button-next hidden lg:block ]"></div>
</Wrapper>

<style lang="scss">
    .alliance-reel {
        position: relative;
    }

    .swiper-button-prev,
    .swiper-button-next {
        --swiper-navigation-color: var(--highlight);
        --swiper-navigation-sides-offset: -46px;

        opacity: 0;
    }

    .swiper-initialized ~ .swiper-button-prev,
    .swiper-initialized ~ .swiper-button-next {
        opacity: 1;
    }

    .swiper-initialized ~ .swiper-button-next.swiper-button-disabled,
    .swiper-initialized ~ .swiper-button-prev.swiper-button-disabled {
        opacity: 0;
    }

    .swiper-button-next.swiper-button-disabled,
    .swiper-button-prev.swiper-button-disabled {
        --swiper-navigation-color: var(--fleet-yellow);
        opacity: 1;
    }

    .swiper-button-prev {
        transform: translateY(-1.25rem);
    }

    .swiper-button-next {
        transform: translateY(-1.25rem);
    }

    .swiper-slide {
        width: min(100%, 700px);
    }
</style>