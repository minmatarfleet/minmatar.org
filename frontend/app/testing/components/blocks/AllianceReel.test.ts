import { experimental_AstroContainer as AstroContainer } from "astro/container";
import { expect, test } from "vitest";
import AllianceReel from "@components/blocks/AllianceReel.astro";

test("AllianceReel defaults", async () => {
  const container = await AstroContainer.create();
  const result = await container.renderToString(AllianceReel, {});

  expect(result).toMatchSnapshot();
});
