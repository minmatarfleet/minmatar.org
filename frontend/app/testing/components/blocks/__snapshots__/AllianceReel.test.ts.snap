// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AllianceReel defaults 1`] = `
"<div class="wrapper [ alliance-reel ]" data-astro-cid-krrzwzg5="true" data-mod data-astro-cid-yv35cmqx style="--wrapper_padding_inline: 0;--wrapper_padding_inline_end: 0;--wrapper_padding_block: 0;--wrapper_max_width: 100%;">  <div class="[ swiper ][ w-full pb-10 ]" id="corporations-list" data-astro-cid-krrzwzg5> <div class="flex-inline [ swiper-wrapper ][ select-none ]" x-data="{
        swiper: null,
        corporation_pagination(index) {
            return this.pagination_icons[index]
        },
        init() {
            this.swiper = new Swiper('.swiper', {
                initialSlide: 0,
                slidesPerView: &#34;auto&#34;,
                spaceBetween: 40,
                pagination: {&#34;clickable&#34;:true,&#34;dynamicBullets&#34;:true,&#34;el&#34;:&#34;.swiper-pagination&#34;},
                loop: false,
                fadeEffect: {
                    crossFade: true
                },
                navigation: {
                    nextEl: '.swiper-button-next',
                    prevEl: '.swiper-button-prev',
                },
            });
        }
    }" data-astro-cid-okd446ho="true" data-astro-cid-ud5fqf5k style="--flex_inline_gap: 0;--flex_inline_justification: flex-start;">   <div class="[ swiper-slide ]" style="margin-right: 40px" data-astro-cid-krrzwzg5> <div class="wrapper [ card component ] [ relative animate-fade-in-up ] relative animate-fade-in-up [ animate-delay-0 ] light-transparency" data-astro-cid-hkp4z2gm="true" data-mod data-astro-cid-yv35cmqx style="--wrapper_padding_inline: var(--space-l-xl);--wrapper_padding_inline_end: var(--space-l-xl);--wrapper_padding_block: var(--component-padding-block);--wrapper_max_width: 100%;">   <img class="[ z-0 ]" src="/images/corporations-tile-background.jpg" data-astro-cid-hkp4z2gm> <div class="block-list [ relative z-10 !justify-between min-h-[340px] ]" data-astro-cid-hkp4z2gm="true" data-astro-cid-s4x3vzov style="--block_list_gap: var(--space-s);">  <div class="flex-inline" slot="header" data-astro-cid-krrzwzg5="true" data-astro-cid-ud5fqf5k style="--flex_inline_gap: var(--space-2xs);--flex_inline_justification: flex-start;">  <img loading="lazy" src="/images/icons/48px-CorporationIcon.png" width="48" height="48" alt="EVE Online Corporation Icon"> <h2 data-astro-cid-krrzwzg5>Corporations</h2>  </div>   <div class="flex-block gap---space-m" data-astro-cid-hkp4z2gm="true" data-mod style="--flex_block_gap: var(--space-m);">    <p data-astro-cid-krrzwzg5>Strong corporations build strong alliances. Learn more about the corporations, CEOs, and membership that make up the Minmatar Fleet Alliance. Perhaps you can submit an application of your own!</p> <div class="flex-inline" data-astro-cid-krrzwzg5="true" data-astro-cid-ud5fqf5k style="--flex_inline_gap: var(--space-s);--flex_inline_justification: flex-start;">  <a href="/alliance/corporations/list/" data-astro-cid-krrzwzg5="true" data-size="sm" style="--button_background_color: var(--fleet-red); --button_color: var(--fleet-yellow)" data-astro-cid-r5e2wo56="true" class="[ button ]"> <div class="flex-inline" data-astro-cid-r5e2wo56="true" data-astro-cid-ud5fqf5k style="--flex_inline_gap: var(--space-xs);--flex_inline_justification: center;">   View  </div>   </a>   </div>     </div>   </div>     </div>    </div> <div class="[ swiper-slide ]" style="margin-right: 40px" data-astro-cid-krrzwzg5> <div class="wrapper [ card component ] [ relative animate-fade-in-up ] relative animate-fade-in-up [ animate-delay-200 ] light-transparency" data-astro-cid-hkp4z2gm="true" data-mod data-astro-cid-yv35cmqx style="--wrapper_padding_inline: var(--space-l-xl);--wrapper_padding_inline_end: var(--space-l-xl);--wrapper_padding_block: var(--component-padding-block);--wrapper_max_width: 100%;">   <img class="[ z-0 ]" src="/images/sigs-tile-background.jpg" data-astro-cid-hkp4z2gm> <div class="block-list [ relative z-10 !justify-between min-h-[340px] ]" data-astro-cid-hkp4z2gm="true" data-astro-cid-s4x3vzov style="--block_list_gap: var(--space-s);">  <div class="flex-inline" slot="header" data-astro-cid-krrzwzg5="true" data-astro-cid-ud5fqf5k style="--flex_inline_gap: var(--space-2xs);--flex_inline_justification: flex-start;">  <img loading="lazy" src="/images/icons/48px-Grouplist.png" width="48" height="48" alt="EVE Online Group List Icon"> <h2 data-astro-cid-krrzwzg5>Special Interest Groups</h2>  </div>   <div class="flex-block gap---space-m" data-astro-cid-hkp4z2gm="true" data-mod style="--flex_block_gap: var(--space-m);">    <p data-astro-cid-krrzwzg5>The purpose of SIGs is to maintain closeness in a massive sea of pilots, as described in <a target="_blank" rel="noreferrer noopener nofollow" href="https://docs.google.com/presentation/d/12W-SQUbKAVssdTkM9ixTm_uTkJiDKCXdF6SA2O0B1PQ/edit?usp=sharing">Town Hall #6</a>. Ultimately, we need to expand and create ways to maintain a close-knit feel in a large organization.</p> <div class="flex-inline" data-astro-cid-krrzwzg5="true" data-astro-cid-ud5fqf5k style="--flex_inline_gap: var(--space-s);--flex_inline_justification: flex-start;">  <a href="/alliance/sigs/list/" data-astro-cid-krrzwzg5="true" data-size="sm" style="--button_background_color: var(--fleet-red); --button_color: var(--fleet-yellow)" data-astro-cid-r5e2wo56="true" class="[ button ]"> <div class="flex-inline" data-astro-cid-r5e2wo56="true" data-astro-cid-ud5fqf5k style="--flex_inline_gap: var(--space-xs);--flex_inline_justification: center;">   View  </div>   </a>   </div>     </div>   </div>     </div>    </div> <div class="[ swiper-slide ]" data-astro-cid-krrzwzg5> <div class="wrapper [ card component ] [ relative animate-fade-in-up ] relative animate-fade-in-up [ animate-delay-400 ] light-transparency" data-astro-cid-hkp4z2gm="true" data-mod data-astro-cid-yv35cmqx style="--wrapper_padding_inline: var(--space-l-xl);--wrapper_padding_inline_end: var(--space-l-xl);--wrapper_padding_block: var(--component-padding-block);--wrapper_max_width: 100%;">   <img class="[ z-0 ]" src="/images/teams-tile-background.jpg" data-astro-cid-hkp4z2gm> <div class="block-list [ relative z-10 !justify-between min-h-[340px] ]" data-astro-cid-hkp4z2gm="true" data-astro-cid-s4x3vzov style="--block_list_gap: var(--space-s);">  <div class="flex-inline" slot="header" data-astro-cid-krrzwzg5="true" data-astro-cid-ud5fqf5k style="--flex_inline_gap: var(--space-2xs);--flex_inline_justification: flex-start;">  <img loading="lazy" src="/images/icons/48px-Augmentations.png" width="48" height="48" alt="EVE Online Augmentation Icon"> <h2 data-astro-cid-krrzwzg5>Teams</h2>  </div>   <div class="flex-block gap---space-m" data-astro-cid-hkp4z2gm="true" data-mod style="--flex_block_gap: var(--space-m);">    <p data-astro-cid-krrzwzg5>Teams are similar to special interest groups, but without them our alliance would cease to exist. These are people running things behind the curtain to ensure that weâ€™re not going to failcascade in 6 months.</p> <div class="flex-inline" data-astro-cid-krrzwzg5="true" data-astro-cid-ud5fqf5k style="--flex_inline_gap: var(--space-s);--flex_inline_justification: flex-start;">  <a href="/alliance/teams/list/" data-astro-cid-krrzwzg5="true" data-size="sm" style="--button_background_color: var(--fleet-red); --button_color: var(--fleet-yellow)" data-astro-cid-r5e2wo56="true" class="[ button ]"> <div class="flex-inline" data-astro-cid-r5e2wo56="true" data-astro-cid-ud5fqf5k style="--flex_inline_gap: var(--space-xs);--flex_inline_justification: center;">   View  </div>   </a>   </div>     </div>   </div>     </div>    </div>   </div>   <div class="swiper-pagination" data-astro-cid-okd446ho></div>  </div> <div class="[ swiper-button-prev hidden lg:block ]" data-astro-cid-krrzwzg5></div> <div class="[ swiper-button-next hidden lg:block ]" data-astro-cid-krrzwzg5></div>  </div>   "
`;
